### ISSUE TITLE
failed to call webhook: Post "https://linkerd-proxy-injector.linkerd.svc:443/?timeout=10s": context deadline exceeded

### ISSUE THREAD

**Issue by joscabmar** [linkerd/linkerd2#9672]
### What is the issue?

Whenever I try to create a new resource in a namespace that does not have the label `config.linkerd.io/admission-webhooks: disabled` I get the following error:

```
Error from server (InternalError): Internal error occurred: failed calling webhook "linkerd-proxy-injector.linkerd.io": failed to call webhook: Post "https://linkerd-proxy-injector.linkerd.svc:443/?timeout=10s": context deadline exceeded
```

### How can it be reproduced?

The override values:
```
deploymentStrategy:
  rollingUpdate:
    maxUnavailable: 1
enablePodAntiAffinity: true
enablePodDisruptionBudget: true
webhookFailurePolicy: Fail
controllerReplicas: 3
identity:
  externalCA: true
  issuer:
    scheme: kubernetes.io/tls
proxyInit:
  iptablesMode: legacy
  runAsRoot: true
enablePSP: true
podMonitor:
  enabled: true
proxyInjector:
  externalSecret: true
  injectCaFrom: linkerd/linkerd-proxy-injector
policyValidator:
  externalSecret: true
  injectCaFrom: linkerd/linkerd-policy-validator
profileValidator:
  externalSecret: true
  injectCaFrom: linkerd/linkerd-sp-validator
```

I omitted `identityTrustAnchorsPEM` but it is also specified.
Installing the chart gives no errors. But just deploying a pod in a namespace where the webhook is active will give you the error:
```
$ kubectl run debugger --rm --tty -i --restart='Never' --namespace linkerd-viz --image tutum/dnsutils --command -- /bin/bash
Error from server (InternalError): Internal error occurred: failed calling webhook "linkerd-proxy-injector.linkerd.io": failed to call webhook: Post "https://linkerd-proxy-injector.linkerd.svc:443/?timeout=10s": context deadline exceeded
```

### Logs, error output, etc

Immediate output from kubectl:
```
Error from server (InternalError): Internal error occurred: failed calling webhook "linkerd-proxy-injector.linkerd.io": failed to call webhook: Post "https://linkerd-proxy-injector.linkerd.svc:443/?timeout=10s": context deadline exceeded
```
When forcing the errors, I see no logs in the linkerd namespace
![Screenshot 2022-10-21 at 13 54 10](https://user-images.githubusercontent.com/6691911/197189936-41848057-4a1a-4159-bcb9-677cebff1d24.png)

### output of `linkerd check -o short`
Linkerd core checks
===================

linkerd-identity
----------------
‼ issuer cert is valid for at least 60 days
    issuer certificate will expire on 2022-10-22T15:26:41Z
    see https://linkerd.io/2.12/checks/#l5d-identity-issuer-cert-not-expiring-soon for hints

linkerd-webhooks-and-apisvc-tls
-------------------------------
‼ proxy-injector cert is valid for at least 60 days
    certificate will expire on 2022-10-21T15:26:41Z
    see https://linkerd.io/2.12/checks/#l5d-proxy-injector-webhook-cert-not-expiring-soon for hints
‼ sp-validator cert is valid for at least 60 days
    certificate will expire on 2022-10-21T15:26:41Z
    see https://linkerd.io/2.12/checks/#l5d-sp-validator-webhook-cert-not-expiring-soon for hints
‼ policy-validator cert is valid for at least 60 days
    certificate will expire on 2022-10-21T15:26:42Z
    see https://linkerd.io/2.12/checks/#l5d-policy-validator-webhook-cert-not-expiring-soon for hints

### Environment

- Kubernetes 1.22
- EKS
- Linkerd stable-2.12.2

### Additional context

The linkerd core checks seem ok. The certificates are renewed on a daily basis via cert-manager.

### Would you like to work on fixing this bug?

yes

**Comment by joscabmar** [comment]
The linkerd core checks seem ok. The certificates are renewed on a daily basis via cert-manager.

**Comment by alpeb** [comment]
Can you confirm the `caBundle` field is getting properly injected into the `linkerd-proxy-injector-webhook-config` MutatingWebhookConfiguration by cert-manager?

**Comment by joscabmar** [comment]
Yes, I can confirm it is present:
```
apiVersion: admissionregistration.k8s.io/v1
kind: MutatingWebhookConfiguration
metadata:
  annotations:
    cert-manager.io/inject-ca-from: linkerd/linkerd-proxy-injector
    meta.helm.sh/release-name: linkerd-control-plane
    meta.helm.sh/release-namespace: linkerd
  creationTimestamp: "2022-10-21T10:22:29Z"
  generation: 4
  labels:
    app.kubernetes.io/managed-by: Helm
    linkerd.io/control-plane-component: proxy-injector
    linkerd.io/control-plane-ns: linkerd
  name: linkerd-proxy-injector-webhook-config
  resourceVersion: "26944686"
  uid: 83a22566-0a82-4ebc-8715-670cdb9659b3
webhooks:
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    caBundle: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJqVENDQVRPZ0F3SUJBZ0lRUlhIbTRpS3k2OWkvR254WkpKS3JoekFLQmdncWhrak9QUVFEQWpBbE1TTXcKSVFZRFZRUURFeHB5YjI5MExteHBibXRsY21RdVkyeDFjM1JsY2k1c2IyTmhiREFlRncweU1qRXdNVE14TURNMApORE5hRncwek1qRXdNVEF4TURNME5ETmFNQ1V4SXpBaEJnTlZCQU1UR25KdmIzUXViR2x1YTJWeVpDNWpiSFZ6CmRHVnlMbXh2WTJGc01Ga3dFd1lIS29aSXpqMENBUVlJS29aSXpqMERBUWNEUWdBRXVLeGNDVnFuUHdWeW9hdlkKWE9pa3JuYjg2ekNpVk9zZTVFZGp1YnJzQjFrNnQ0MzVxSTFFK1FGcTI2MGVoYXhGeGw0bTJrVmsrUG1xejNIegpDVW1oSHFORk1FTXdEZ1lEVlIwUEFRSC9CQVFEQWdFR01CSUdBMVVkRXdFQi93UUlNQVlCQWY4Q0FRRXdIUVlEClZSME9CQllFRkg2bmV4TGNCWWw0bWw1alRmNlNlQXVYS1E5OE1Bb0dDQ3FHU000OUJBTUNBMGdBTUVVQ0lRRG8KUVJMdVNVWTNXN1JTWTE4RTNLSGxIQVJuK0xDQ2k0U3Z2bHBWY3RQcHV3SWdhZXkxUUtUa0RtQTVxS1NPaE91aApBVXR6Mm9INm9BMVl3YjlKaEJ0cG85az0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
    service:
      name: linkerd-proxy-injector
      namespace: linkerd
      path: /
      port: 443
  failurePolicy: Fail
  matchPolicy: Equivalent
  name: linkerd-proxy-injector.linkerd.io
  namespaceSelector:
    matchExpressions:
    - key: config.linkerd.io/admission-webhooks
      operator: NotIn
      values:
      - disabled
    - key: kubernetes.io/metadata.name
      operator: NotIn
      values:
      - kube-system
      - cert-manager
  objectSelector:
    matchExpressions:
    - key: linkerd.io/control-plane-component
      operator: DoesNotExist
    - key: linkerd.io/cni-resource
      operator: DoesNotExist
  reinvocationPolicy: Never
  rules:
  - apiGroups:
    - ""
    apiVersions:
    - v1
    operations:
    - CREATE
    resources:
    - pods
    - services
    scope: '*'
  sideEffects: None
  timeoutSeconds: 10
```

ca-injection does not seem to be the problem. The CA that I used was root.linkerd.cluster.local for all the webhook certificates and to generate the identity CA as well. This is because I have already tried with an intermediate CA for `webhook.linkerd.cluster.local` which did not seem to work. Then I tried with another root CA for `webhook.linkerd.cluster.local`, but it still did not work. So I made it this way to simplify it and make it coincide with the `linkerd-identity-trust-roots`.

**Comment by joscabmar** [comment]
I have tried in a fresh installed EKS cluster with:

Kubernetes 1.22 (it still uses docker container runtime in EKS)
coredns v1.8.4-eksbuild.2 
kube-proxy v1.22.11-eksbuild.2
vpc-cni v1.11.4-eksbuild.1

And I ran via CLI:
```
linkerd check
linkerd check --pre
linkerd install --crds | kubectl apply -f -
linkerd install --set proxyInit.runAsRoot=true | kubectl apply -f -
linkerd check
# Everything was OK in the check
curl --proto '=https' --tlsv1.2 -sSfL https://run.linkerd.io/emojivoto.yml | kubectl apply -f -
kubectl get -n emojivoto deploy -o yaml | linkerd inject - | kubectl apply -f -
```
And still I can see injection is not working properly:
```
kubectl get all
-A
NAMESPACE     NAME                                          READY   STATUS    RESTARTS   AGE
emojivoto     pod/emoji-696d9d8f95-bch8c                    1/1     Running   0          23m
emojivoto     pod/vote-bot-6d7677bb68-xkf6z                 1/1     Running   0          23m
emojivoto     pod/voting-ff4c54b8d-r8dk4                    1/1     Running   0          23m
emojivoto     pod/web-5f86686c4d-7l4qf                      1/1     Running   0          23m
kube-system   pod/aws-node-5qvdz                            1/1     Running   0          112s
kube-system   pod/aws-node-8kz6b                            1/1     Running   0          60s
kube-system   pod/aws-node-cfx2l                            1/1     Running   0          2m19s
kube-system   pod/aws-node-pc6fs                            1/1     Running   0          86s
kube-system   pod/aws-node-v2xw4                            1/1     Running   0          2m45s
kube-system   pod/aws-node-vb9c7                            1/1     Running   0          27s
kube-system   pod/coredns-878fc65df-4b9j4                   1/1     Running   0          2m16s
kube-system   pod/coredns-878fc65df-rqctf                   1/1     Running   0          2m16s
kube-system   pod/ebs-csi-controller-dc5c66d96-2rpml        6/6     Running   0          90m
kube-system   pod/ebs-csi-controller-dc5c66d96-5nvrk        6/6     Running   0          90m
kube-system   pod/ebs-csi-node-26s7k                        3/3     Running   0          90m
kube-system   pod/ebs-csi-node-4mxc4                        3/3     Running   0          90m
kube-system   pod/ebs-csi-node-52zn6                        3/3     Running   0          90m
kube-system   pod/ebs-csi-node-8tsxt                        3/3     Running   0          90m
kube-system   pod/ebs-csi-node-b9cm4                        3/3     Running   0          90m
kube-system   pod/ebs-csi-node-rcgf8                        3/3     Running   0          90m
kube-system   pod/kube-proxy-7dwgg                          1/1     Running   0          2m22s
kube-system   pod/kube-proxy-9g4q8                          1/1     Running   0          2m32s
kube-system   pod/kube-proxy-bv97x                          1/1     Running   0          2m8s
kube-system   pod/kube-proxy-c9lw9                          1/1     Running   0          2m27s
kube-system   pod/kube-proxy-hqrrf                          1/1     Running   0          2m12s
kube-system   pod/kube-proxy-wwpvx                          1/1     Running   0          2m17s
linkerd       pod/linkerd-destination-5695446dfc-7gckg      4/4     Running   0          25m
linkerd       pod/linkerd-identity-6db6d66989-9znvg         2/2     Running   0          25m
linkerd       pod/linkerd-proxy-injector-54b66cbc77-pr982   2/2     Running   0          25m

NAMESPACE     NAME                                TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE
default       service/kubernetes                  ClusterIP   172.20.0.1       <none>        443/TCP             98m
emojivoto     service/emoji-svc                   ClusterIP   172.20.97.158    <none>        8080/TCP,8801/TCP   24m
emojivoto     service/voting-svc                  ClusterIP   172.20.27.181    <none>        8080/TCP,8801/TCP   24m
emojivoto     service/web-svc                     ClusterIP   172.20.154.9     <none>        80/TCP              23m
kube-system   service/kube-dns                    ClusterIP   172.20.0.10      <none>        53/UDP,53/TCP       98m
linkerd       service/linkerd-dst                 ClusterIP   172.20.32.13     <none>        8086/TCP            25m
linkerd       service/linkerd-dst-headless        ClusterIP   None             <none>        8086/TCP            25m
linkerd       service/linkerd-identity            ClusterIP   172.20.33.153    <none>        8080/TCP            25m
linkerd       service/linkerd-identity-headless   ClusterIP   None             <none>        8080/TCP            25m
linkerd       service/linkerd-policy              ClusterIP   None             <none>        8090/TCP            25m
linkerd       service/linkerd-policy-validator    ClusterIP   172.20.126.93    <none>        443/TCP             25m
linkerd       service/linkerd-proxy-injector      ClusterIP   172.20.179.181   <none>        443/TCP             25m
linkerd       service/linkerd-sp-validator        ClusterIP   172.20.81.181    <none>        443/TCP             25m

NAMESPACE     NAME                                  DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR              AGE
kube-system   daemonset.apps/aws-node               6         6         6       6            6           <none>                     98m
kube-system   daemonset.apps/ebs-csi-node           6         6         6       6            6           kubernetes.io/os=linux     90m
kube-system   daemonset.apps/ebs-csi-node-windows   0         0         0       0            0           kubernetes.io/os=windows   90m
kube-system   daemonset.apps/kube-proxy             6         6         6       6            6           <none>                     98m

NAMESPACE     NAME                                     READY   UP-TO-DATE   AVAILABLE   AGE
emojivoto     deployment.apps/emoji                    1/1     1            1           23m
emojivoto     deployment.apps/vote-bot                 1/1     1            1           23m
emojivoto     deployment.apps/voting                   1/1     1            1           23m
emojivoto     deployment.apps/web                      1/1     1            1           23m
kube-system   deployment.apps/coredns                  2/2     2            2           98m
kube-system   deployment.apps/ebs-csi-controller       2/2     2            2           90m
linkerd       deployment.apps/linkerd-destination      1/1     1            1           25m
linkerd       deployment.apps/linkerd-identity         1/1     1            1           25m
linkerd       deployment.apps/linkerd-proxy-injector   1/1     1            1           25m

NAMESPACE     NAME                                                DESIRED   CURRENT   READY   AGE
emojivoto     replicaset.apps/emoji-66ccdb4d86                    0         0         0       23m
emojivoto     replicaset.apps/emoji-696d9d8f95                    1         1         1       23m
emojivoto     replicaset.apps/vote-bot-69754c864f                 0         0         0       23m
emojivoto     replicaset.apps/vote-bot-6d7677bb68                 1         1         1       23m
emojivoto     replicaset.apps/voting-f999bd4d7                    0         0         0       23m
emojivoto     replicaset.apps/voting-ff4c54b8d                    1         1         1       23m
emojivoto     replicaset.apps/web-5f86686c4d                      1         1         1       23m
emojivoto     replicaset.apps/web-79469b946f                      0         0         0       23m
kube-system   replicaset.apps/coredns-5947f47f5f                  0         0         0       98m
kube-system   replicaset.apps/coredns-878fc65df                   2         2         2       2m16s
kube-system   replicaset.apps/ebs-csi-controller-dc5c66d96        2         2         2       90m
linkerd       replicaset.apps/linkerd-destination-5695446dfc      1         1         1       25m
linkerd       replicaset.apps/linkerd-identity-6db6d66989         1         1         1       25m
linkerd       replicaset.apps/linkerd-proxy-injector-54b66cbc77   1         1         1       25m

NAMESPACE   NAME                              SCHEDULE      SUSPEND   ACTIVE   LAST SCHEDULE   AGE
linkerd     cronjob.batch/linkerd-heartbeat   54 12 * * *   False     0        16m             25m
```

Linkerd checks:
```
$ linkerd check
Linkerd core checks
===================

kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ control plane pods are ready
√ cluster networks contains all pods
√ cluster networks contains all services

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
√ control plane CustomResourceDefinitions exist
√ control plane MutatingWebhookConfigurations exist
√ control plane ValidatingWebhookConfigurations exist
√ proxy-init container runs as root user if docker container runtime is used

linkerd-identity
----------------
√ certificate config is valid
√ trust anchors are using supported crypto algorithm
√ trust anchors are within their validity period
√ trust anchors are valid for at least 60 days
√ issuer cert is using supported crypto algorithm
√ issuer cert is within its validity period
√ issuer cert is valid for at least 60 days
√ issuer cert is issued by the trust anchor

linkerd-webhooks-and-apisvc-tls
-------------------------------
√ proxy-injector webhook has valid cert
√ proxy-injector cert is valid for at least 60 days
√ sp-validator webhook has valid cert
√ sp-validator cert is valid for at least 60 days
√ policy-validator webhook has valid cert
√ policy-validator cert is valid for at least 60 days

linkerd-version
---------------
√ can determine the latest version
√ cli is up-to-date

control-plane-version
---------------------
√ can retrieve the control plane version
√ control plane is up-to-date
√ control plane and cli versions match

linkerd-control-plane-proxy
---------------------------
√ control plane proxies are healthy
√ control plane proxies are up-to-date
√ control plane proxies and cli versions match

Status check results are √

$ linkerd -n emojivoto check --proxy
Linkerd core checks
===================

kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ control plane pods are ready
√ cluster networks contains all pods
√ cluster networks contains all services

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
√ control plane CustomResourceDefinitions exist
√ control plane MutatingWebhookConfigurations exist
√ control plane ValidatingWebhookConfigurations exist
√ proxy-init container runs as root user if docker container runtime is used

linkerd-identity
----------------
√ certificate config is valid
√ trust anchors are using supported crypto algorithm
√ trust anchors are within their validity period
√ trust anchors are valid for at least 60 days
√ issuer cert is using supported crypto algorithm
√ issuer cert is within its validity period
√ issuer cert is valid for at least 60 days
√ issuer cert is issued by the trust anchor

linkerd-webhooks-and-apisvc-tls
-------------------------------
√ proxy-injector webhook has valid cert
√ proxy-injector cert is valid for at least 60 days
√ sp-validator webhook has valid cert
√ sp-validator cert is valid for at least 60 days
√ policy-validator webhook has valid cert
√ policy-validator cert is valid for at least 60 days

linkerd-identity-data-plane
---------------------------
√ data plane proxies certificate match CA

linkerd-version
---------------
√ can determine the latest version
√ cli is up-to-date

linkerd-control-plane-proxy
---------------------------
√ control plane proxies are healthy
√ control plane proxies are up-to-date
√ control plane proxies and cli versions match

linkerd-data-plane
------------------
√ data plane namespace exists
× data plane proxies are ready
    no "linkerd-proxy" containers found in the "emojivoto" namespace
    see https://linkerd.io/2.12/checks/#l5d-data-plane-ready for hints

Status check results are ×
```

Logs in linkerd-proxy-injector pod: https://pastebin.com/gR1VnMM8
Logs in linkerd-destination pod:  https://pastebin.com/3bZpHTmK
Logs in linkerd-identity pod: https://pastebin.com/bUEht5Ju

It showed some DNS resolution errors, but when I deployed a dnsutils pod in that namespace and tested that DNS was actually working, it did.

**Comment by alpeb** [comment]
@joscabmar while trying to repro your last series of steps, I pasted the last couple of steps as is, which didn't give a chance to k8s to persist the initial workloads, and the last step failed which resulted in the pods not getting injected. Is by any chance what happened to you as well and you didn't notice the error on the command line?
```bash
$ curl --proto '=https' --tlsv1.2 -sSfL https://run.linkerd.io/emojivoto.yml | kubectl apply -f -
namespace/emojivoto created                                                                                                                                                                                                                    serviceaccount/emoji created                                                                                                                                                                                                                   serviceaccount/voting created                                                                                                                                                                                                                  serviceaccount/web created                                                                                                                                                                                                                     
service/emoji-svc created                                  
service/voting-svc created                                                                                             
service/web-svc created               
deployment.apps/emoji created                                                                                                                                                                                                                  
deployment.apps/vote-bot created                        
deployment.apps/voting created                                                                                                                                                                                                                 deployment.apps/web created                                                                                                                                                                                                                                                                                                                                                                                                                                                                   deployment "emoji" injected                                                                                                                                                                                                                    deployment "vote-bot" injected                                                                                                                                                                                                                 deployment "voting" injected                                                                                                                                                                                                                   deployment "web" injected         

$ kubectl get -n emojivoto deploy -o yaml | linkerd inject - | kubectl apply -f -
Error from server (Conflict): error when applying patch:
{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"apps/v1\",\"kind\":\"Deployment\",\"metadata\":{\"annotations\":{\"deployment.kubernetes.io/revision\":\"1\"},\"generation\":1,\"labels\":{\"app.kubernetes.io/name\":\"emoji\",\"app.kubernetes.io/part-of\":\"emojivoto\",\"app.kubernetes.io/version\":\"v11\"},\"name\":\"emoji\",\"namespace\":\"emojivoto\",\"resourceVersion\":\"40112\",\"uid\":\"17e59679-6225-42ed-8660-97a451bc2b76\"},\"spec\":{\"progressDeadlineSeconds\":600,\"replicas\":1,\"revisionHistoryLimit\":10,\"selector\":{\"matchLabels\":{\"app\":\"emoji-svc\",\"version\":\"v11\"}},\"strategy\":{\"rollingUpdate\":{\"maxSurge\":\"25%\",\"maxUnavailable\":\"25%\"},\"type\":\"RollingUpdate\"},\"template\":{\"metadata\":{\"annotations\":{\"linkerd.io/inject\":\"enabled\"},\"labels\":{\"app\":\"emoji-svc\",\"version\":\"v11\"}},\"spec\":{\"containers\":[{\"env\":[{\"name\":\"GRPC_PORT\",\"value\":\"8080\"},{\"name\":\"PROM_PORT\",\"value\":\"8801\"}],\"image\":\"docker.l5d.io/buoyantio/emojivoto-emoji-svc:v11\",\"imagePullPolicy\":\"IfNotPresent\",\"name\":\"emoji-svc\",\"ports\":[{\"containerPort\":8080,\"name\":\"grpc\",\"protocol\":\"TCP\"},{\"containerPort\":8801,\"name\":\"prom\",\"protocol\":\"TCP\"}],\"resources\":{\"requests\":{\"cpu\":\"100m\"}},\"terminationMessagePath\":\"/dev/termination-log\",\"terminationMessagePolicy\":\"File\"}],\"dnsPolicy\":\"ClusterFirst\",\"restartPolicy\":\"Always\",\"schedulerName\":\"default-scheduler\",\"securityContext\":{},\"serviceAccount\":\"emoji\",\"serviceAccountName\":\"emoji\",\"terminationGracePeriodSeconds\":30}}},\"status\":{\"conditions\":[{\"message\":\"Deployment does not have minimum availability.\",\"reason\":\"MinimumReplicasUnavailable\",\"status\":\"False\",\"type\":\"Available\"},{\"message\":\"ReplicaSet \\\"emoji-5dbdd567bd\\\" is progressing.\",\"reason\":\"ReplicaSetUpdated\",\"status\":\"True\",\"type\":\"Progressing\"}],\"observedGeneration\":1,\"replicas\":1,\"unavailableReplicas\":1,\"updatedReplicas\":1}}\n"},"resourceVersion":"40112"},"spec":{"template":{"metadata":{"annotations":{"linkerd.io/inject":"enabled"}}}},"status":{"$setElementOrder/conditions":[{"type":"Available"},{"type":"Progressing"}],"conditions":[{"message":"Deployment does not have minimum availability.","reason":"MinimumReplicasUnavailable","status":"False","type":"Available"},{"message":"ReplicaSet \"emoji-5dbdd567bd\" is progressing.","reason":"ReplicaSetUpdated","type":"Progressing"}],"unavailableReplicas":1}}
to:
Resource: "apps/v1, Resource=deployments", GroupVersionKind: "apps/v1, Kind=Deployment"
Name: "emoji", Namespace: "emojivoto"
for: "STDIN": error when patching "STDIN": Operation cannot be fulfilled on deployments.apps "emoji": the object has been modified; please apply your changes to the latest version and try again

```

**Comment by joscabmar** [comment]
Hi @alpeb! Thank you for your response! No, I did not get that error message when running the commands.

**Comment by alpeb** [comment]
@joscabmar could you please provide me with these two extra pieces of information?

- Logs for the `proxy-injector` container in the `linkerd-proxy-injector` pod (you can bounce an emojivoto pod and see what log entries this produces here)
- Description of the emoji deployment, in particular to check its events and see if there was a reason for the injector to skip its injection.

**Comment by alpeb** [comment]
Also, I just noticed in your minimal repro above that the aws-node, core-dns and kube-proxy pods are younger than emojivoto's and linkerd's; any idea how could that happened?

```
kubectl get all
-A
NAMESPACE     NAME                                          READY   STATUS    RESTARTS   AGE
emojivoto     pod/emoji-696d9d8f95-bch8c                    1/1     Running   0          23m
emojivoto     pod/vote-bot-6d7677bb68-xkf6z                 1/1     Running   0          23m
emojivoto     pod/voting-ff4c54b8d-r8dk4                    1/1     Running   0          23m
emojivoto     pod/web-5f86686c4d-7l4qf                      1/1     Running   0          23m
kube-system   pod/aws-node-5qvdz                            1/1     Running   0          112s
kube-system   pod/aws-node-8kz6b                            1/1     Running   0          60s
kube-system   pod/aws-node-cfx2l                            1/1     Running   0          2m19s
kube-system   pod/aws-node-pc6fs                            1/1     Running   0          86s
kube-system   pod/aws-node-v2xw4                            1/1     Running   0          2m45s
kube-system   pod/aws-node-vb9c7                            1/1     Running   0          27s
kube-system   pod/coredns-878fc65df-4b9j4                   1/1     Running   0          2m16s
kube-system   pod/coredns-878fc65df-rqctf                   1/1     Running   0          2m16s
kube-system   pod/ebs-csi-controller-dc5c66d96-2rpml        6/6     Running   0          90m
kube-system   pod/ebs-csi-controller-dc5c66d96-5nvrk        6/6     Running   0          90m
kube-system   pod/ebs-csi-node-26s7k                        3/3     Running   0          90m
kube-system   pod/ebs-csi-node-4mxc4                        3/3     Running   0          90m
kube-system   pod/ebs-csi-node-52zn6                        3/3     Running   0          90m
kube-system   pod/ebs-csi-node-8tsxt                        3/3     Running   0          90m
kube-system   pod/ebs-csi-node-b9cm4                        3/3     Running   0          90m
kube-system   pod/ebs-csi-node-rcgf8                        3/3     Running   0          90m
kube-system   pod/kube-proxy-7dwgg                          1/1     Running   0          2m22s
kube-system   pod/kube-proxy-9g4q8                          1/1     Running   0          2m32s
kube-system   pod/kube-proxy-bv97x                          1/1     Running   0          2m8s
kube-system   pod/kube-proxy-c9lw9                          1/1     Running   0          2m27s
kube-system   pod/kube-proxy-hqrrf                          1/1     Running   0          2m12s
kube-system   pod/kube-proxy-wwpvx                          1/1     Running   0          2m17s
linkerd       pod/linkerd-destination-5695446dfc-7gckg      4/4     Running   0          25m
linkerd       pod/linkerd-identity-6db6d66989-9znvg         2/2     Running   0          25m
linkerd       pod/linkerd-proxy-injector-54b66cbc77-pr982   2/2     Running   0          25m
```

**Comment by joscabmar** [comment]
> Also, I just noticed in your minimal repro above that the aws-node, core-dns and kube-proxy pods are younger than emojivoto's and linkerd's; any idea how could that happened?
> 
> ```
> kubectl get all
> -A
> NAMESPACE     NAME                                          READY   STATUS    RESTARTS   AGE
> emojivoto     pod/emoji-696d9d8f95-bch8c                    1/1     Running   0          23m
> emojivoto     pod/vote-bot-6d7677bb68-xkf6z                 1/1     Running   0          23m
> emojivoto     pod/voting-ff4c54b8d-r8dk4                    1/1     Running   0          23m
> emojivoto     pod/web-5f86686c4d-7l4qf                      1/1     Running   0          23m
> kube-system   pod/aws-node-5qvdz                            1/1     Running   0          112s
> kube-system   pod/aws-node-8kz6b                            1/1     Running   0          60s
> kube-system   pod/aws-node-cfx2l                            1/1     Running   0          2m19s
> kube-system   pod/aws-node-pc6fs                            1/1     Running   0          86s
> kube-system   pod/aws-node-v2xw4                            1/1     Running   0          2m45s
> kube-system   pod/aws-node-vb9c7                            1/1     Running   0          27s
> kube-system   pod/coredns-878fc65df-4b9j4                   1/1     Running   0          2m16s
> kube-system   pod/coredns-878fc65df-rqctf                   1/1     Running   0          2m16s
> kube-system   pod/ebs-csi-controller-dc5c66d96-2rpml        6/6     Running   0          90m
> kube-system   pod/ebs-csi-controller-dc5c66d96-5nvrk        6/6     Running   0          90m
> kube-system   pod/ebs-csi-node-26s7k                        3/3     Running   0          90m
> kube-system   pod/ebs-csi-node-4mxc4                        3/3     Running   0          90m
> kube-system   pod/ebs-csi-node-52zn6                        3/3     Running   0          90m
> kube-system   pod/ebs-csi-node-8tsxt                        3/3     Running   0          90m
> kube-system   pod/ebs-csi-node-b9cm4                        3/3     Running   0          90m
> kube-system   pod/ebs-csi-node-rcgf8                        3/3     Running   0          90m
> kube-system   pod/kube-proxy-7dwgg                          1/1     Running   0          2m22s
> kube-system   pod/kube-proxy-9g4q8                          1/1     Running   0          2m32s
> kube-system   pod/kube-proxy-bv97x                          1/1     Running   0          2m8s
> kube-system   pod/kube-proxy-c9lw9                          1/1     Running   0          2m27s
> kube-system   pod/kube-proxy-hqrrf                          1/1     Running   0          2m12s
> kube-system   pod/kube-proxy-wwpvx                          1/1     Running   0          2m17s
> linkerd       pod/linkerd-destination-5695446dfc-7gckg      4/4     Running   0          25m
> linkerd       pod/linkerd-identity-6db6d66989-9znvg         2/2     Running   0          25m
> linkerd       pod/linkerd-proxy-injector-54b66cbc77-pr982   2/2     Running   0          25m
> ```

That usually happens when you update those EKS addons.

**Comment by joscabmar** [comment]
Thank you for your help @alpeb. I was able to solve the issue by replacing the docker runtime by containerd and I also used the latest Helm chart version available: `1.9.5`.
